src/forms/Checkbox.jsx

```diff
@@ -15,17 +15,25 @@ export const DISABLED_CHECKBOX_CLASS = 'disabled';
 class Checkbox extends React.PureComponent {
 	constructor(props) {
 		super(props);
-		this.state = {
-			checked: props.checked || false,
+		const state = {
+			focused: false,
 		};
+
+		if (props.controlled) {
+			state.checked = props.checked || false;
```

Comment:
How about to reduce some code:
const state = {
   focused: false,
   checked: props.controlled ? props.checked : false
};